tests/005-test-devmapper-cleanup

```diff
@@ -0,0 +1,61 @@
+source $SRCDIR/libtest.sh
+
+# Test "docker-storage-setup reset". Returns 0 on success and 1 on failure.
+test_reset_devmapper() {
+  local devs=$DEVS
+  local test_status
+  local testname=`basename "$0"`
+  local vg_name
+  local vg_name="dss-test-foo"
+
+ # Error out if any pre-existing volume group vg named dss-test-foo
+  for vg in $(vgs --noheadings -o vg_name); do
+    if [ "$vg" == "$vg_name" ]; then
+      echo "ERROR: $testname: Volume group $vg_name already exists."
+      return 1
+    fi
+  done
+
+  # Create config file
+  clean_config_files
+  cat << EOF > /etc/sysconfig/docker-storage-setup
+DEVS="$devs"
+VG=$vg_name
+EOF
+
+ # Run docker-storage-setup
+ $DSSBIN >> $LOGS 2>&1
+
+ # Test failed.
+ if [ $? -ne 0 ]; then
+    cleanup $vg_name "$devs"
+    return 1
+ fi
+
+ test_status=1
+ # Make sure volume group $VG got created.
+  for vg in $(vgs --noheadings -o vg_name); do
+    if [ "$vg" == "$vg_name" ]; then
+      test_status=0
+      break
+    fi
+  done
+
+  test_status=0
+  $DSSBIN --reset >> $LOGS 2>&1
+  # Test failed.
+  if [ $? -ne 0 ]; then
+     test_status=1
+  elif [ -e /etc/sysconfig/docker-storage ]; then
```

Comment:
I think we nee to make test_status thing little cleaner. It would be good if test_status is initialized to 1 and we set it to zero once only after test is successful. In all the other cases simple do cleanup and return.

This test case now seems to be a mix. First we initialized test_status=1, then checked presence of volume group and set to 0. And then set to zero again and then setting it to 1 on failure etc.

So please untangle this.